cg/services/order_validation_service/validators/inter_field/rules.py

```diff
@@ -24,3 +29,21 @@ def validate_name_required_for_new_order(order: Order, **kwargs) -> list[OrderEr
         error = OrderNameRequiredError()
         errors.append(error)
     return errors
+
+
+def validate_application_compatibility(
+    order: Order, store: Store, **kwargs
```

Comment:
The order model does not contain cases/samples so this type hint is incorrect! I'd rather we defined this rule for the tomte workflow since we do not have an appropriate base class for orders with cases (unless you want to add such a base class in this PR)